code/datums/components/acid.dm

```diff
@@ -0,0 +1,178 @@
+/** Component representing acid applied to an object.
+  *
+  * Must be attached to an atom.
+  * Processes, repeatedly damaging whatever it is attached to.
+  * If the parent atom is a turf it applies acid to the contents of the turf.
+  */
+/datum/component/acid
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	/// The strength of the acid on the parent object.
+	var/acid_power
+	/// The volume of acid on the parent object.
+	var/acid_volume
+	/// The ambiant sound of acid eating away at the parent object.
+	var/datum/looping_sound/sizzle
+	/// Used exclusively for melting turfs. TODO: Move integrity to the atom level so that this can be dealt with there.
+	var/parent_integrity = 30
+
+/datum/component/acid/Initialize(_acid_power, _acid_volume)
+	if((_acid_power) <= 0 || (_acid_volume <= 0))
+		return COMPONENT_INCOMPATIBLE // Not enough acid.
```

Comment:
Add an informative `stack_trace` message here. This shouldn't happen by default, should it?

Reply:
This should not happen be default, no. I'll make some changes.

Reply:
I've added `WARNING` calls and have prevented the `AddComponent` calls from running under these circumstances. That should be enough.